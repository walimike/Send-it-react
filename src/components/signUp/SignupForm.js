import React from "react";

const SignupForm = props => {
  let Loader = require('react-loader')
  return (
    <div className="row">
      <div className="col s12 15 offset-12">
        <h4 className="center-align regHeader">REGISTER</h4>
        <form onSubmit={props.onSubmit} id="signUpForm" className="form">
          <div className="input-field">
            <input
              type="text"
              id="username"
              name="username"
              value={props.username}
              onChange={props.onChange}
              className="formInput"
              required
            />
            <label className="active" htmlFor="username">
              Username
            </label>
          </div>

          <div className="input-field">
            <label className="active" htmlFor="email">
              Email
            </label>
            <input
              type="email"
              id="email"
              name="email"
              value={props.email}
              onChange={props.onChange}
              className="formInput"
              required
            />
          </div>

          <div className="input-field">
            <label className="active" htmlFor="password">
              Password
            </label>
            <input
              type="password"
              id="password"
              name="password"
              value={props.password}
              onChange={props.onChange}
              className="formInput"
              required
            />
          </div>

          <div className="input-field">
            <label className="active" htmlFor="password2">
              Confirm Password 
            </label>
            <input
              type="password"
              id="password2"
              name="password2"
              value={props.password2}
              onChange={props.onChange}
              required
            />
          </div>

          <div className="input-field center">
            <Loader loaded={!props.isLoading}>
              <button type="submit" className="waves-effect waves-light btn-small regButton">
                Register
              </button>
            </Loader>
          </div>
         
        </form>
      </div>
    
    </div>
  );
};

export default SignupForm;