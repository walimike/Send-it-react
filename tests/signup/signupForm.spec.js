import React from 'react';
import { shallow } from 'enzyme';

import SignupForm from "../../src/components/signUp/SignupForm";


describe('SignupForm', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<SignupForm />);
      expect(wrapper).toMatchSnapshot();
    });
  });