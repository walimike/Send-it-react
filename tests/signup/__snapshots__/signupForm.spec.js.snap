// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm should render without crashing 1`] = `ShallowWrapper {}`;
